### PR TITLE
Initial Creation of Simulator FM

### PR DESCRIPTION
**Pull request by DubStepMad**
Name: Simulator FM
About:
Initial Creation of Simulator FM Presence

image
image
image
image
image
image
image

### CODE REVIEW

File: websites/S/SimulatorFM/presence.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+const presence = new Presence({
+  clientId: "790721299126943744"
+}),
+browsingStamp = Math.floor(Date.now() / 1000),
+dj = document.querySelector(".m-player__presenter-name") as HTMLElement;
+
```
Comment (Bas950) [inline websites/S/SimulatorFM/presence.ts:6]: this element is only get on website load, is that correct?

Comment (DubStepMad) [reply websites/S/SimulatorFM/presence.ts:6]: Sorry for the slow reply, that element should be updated each time as it can change.

Comment (DubStepMad) [reply websites/S/SimulatorFM/presence.ts]: Updated the code to reflect the expected outcome.

**Comment by ririxidev** [comment]
Sadly this site isn't 2 months old. sorry.
image

**Comment by DubStepMad** [comment]
@ririxidev The domain was moved over to OVH as it was with a previous domain name registar.

Evidence can be provided in the form of PPL Invoice (Invoice Music License) and the license terms and agreement with the mentioned domain name.

Also I can provide proof of the previous domain name registar with an invoice.

File: websites/S/SimulatorFM/presence.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+let dj: HTMLElement,
+  title: HTMLElement,
+  artist: HTMLElement,
+  player: HTMLAudioElement;
+
+presence.on("UpdateData", async () => {
+  dj = document.querySelector(".m-player__presenter-name");
+  player = document.querySelector(".fa-pause");
+  title = document.querySelector(".m-player__title");
+  artist = document.querySelector(".m-player__artist");
```
Comment (Bas950) [inline websites/S/SimulatorFM/presence.ts:15]: since the elements arent changed somewhere else just make them const in the updatedata event.

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by Bas950** [comment]
Reopened the PR as proof was received of domain age.

File: websites/S/SimulatorFM/presence.ts (hunk 2 above)
Comment (DubStepMad) [reply websites/S/SimulatorFM/presence.ts]: Sorted as requested.

**Review by Bas950 — APPROVED** [review]

File: websites/S/SimulatorFM/presence.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+presence.on("UpdateData", async () => {
+  const dj = document.querySelector(".m-player__presenter-name") as HTMLElement,
+  player = document.querySelector(".fa-pause") as HTMLElement,
+  title = document.querySelector(".m-player__title") as HTMLElement,
+  artist = document.querySelector(".m-player__artist") as HTMLElement;
+
+  const presenceData: PresenceData = {
+    largeImageKey: "largelogo"
+  };
```
Comment (ririxidev) [inline websites/S/SimulatorFM/presence.ts:14]: merge presenceData with first const :)

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/S/SimulatorFM/presence.ts (hunk 3 above)
Comment (DubStepMad) [reply websites/S/SimulatorFM/presence.ts]: Changes have been made as mentioned above.

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**